hamocc/mo_carchm.F90

```diff
@@ -0,0 +1,880 @@
+    real,    intent(in)  :: ta     ! total alkalinity [eq/kg].
+    real,    intent(in)  :: sit    ! silicate concentration [mol/kg].
+    real,    intent(in)  :: pt     ! phosphate concentration [mol/kg].
+    real,    intent(in)  :: KH     ! equilibrium constant K0  = [H2CO3]/pCO2.
+    real,    intent(in)  :: K1     ! equilibrium constant K1  = [H][HCO3]/[H2CO3].
+    real,    intent(in)  :: K2     ! equilibrium constant K2  = [H][CO3]/[HCO3].
+    real,    intent(in)  :: Kb     ! equilibrium constant Kb  = [H][BO2]/[HBO2].
+    real,    intent(in)  :: Kw     ! equilibrium constant Kw  = [H][OH].
+    real,    intent(in)  :: Ks1    ! equilibrium constant Ks1 = [H][SO4]/[HSO4].
+    real,    intent(in)  :: Kf     ! equilibrium constant Kf  = [H][F]/[HF].
+    real,    intent(in)  :: Ksi    ! equilibrium constant Ksi = [H][SiO(OH)3]/[Si(OH)4].
+    real,    intent(in)  :: K1p    ! equilibrium constant K1p = [H][H2PO4]/[H3PO4].
+    real,    intent(in)  :: K2p    ! equilibrium constant K2p = [H][HPO4]/[H2PO4].
+    real,    intent(in)  :: K3p    ! equilibrium constant K3p = [H][PO4]/[HPO4].
+    real,    intent(out) :: tc_sat ! saturated total DIC concentration [mol/kg].
+    integer, intent(in)  :: niter  ! maximum number of iteration
+
+    ! Parameters to set accuracy of iteration
+    real, parameter :: eps=5.e-5
+
+    ! Local varibles
+    integer :: jit
+    real    :: s,scl,borat,sti,ft
+    real    :: hso4,hf,hsi,hpo4,ab,aw,ah2o,ah2,erel
+    real    :: dic_h2co3,dic_hco3,dic_co3,ah1,ac
+
+    ! Calculate concentrations for borate, sulfate, and fluoride; see Dickson, A.G.,
+    ! Sabine, C.L. and Christian, J.R. (Eds.) 2007. Guide to best practices
+    ! for ocean CO2 measurements. PICES Special Publication 3, chapter 5 p. 10
+    s = MAX(25.,saln)
+    scl = s * salchl
+    borat = bor1 * scl * bor2      ! Uppstrom (1974)
+    sti = 0.14 * scl / 96.062      ! Morris & Riley (1966)
+    ft = 0.000067 * scl / 18.9984  ! Riley (1965)
+    ah1=1.e-8
+    dic_h2co3 = Kh * pco2 * 1e-6
+
+    iflag: do jit = 1,niter
+      hso4 = sti / ( 1. + Ks1 / ( ah1 / ( 1. + sti / Ks1 ) ) )
+      hf   = 1. / ( 1. + Kf / ah1 )
+      hsi  = 1./ ( 1. + ah1 / Ksi )
+      hpo4 = ( K1p * K2p * ( ah1 + 2. * K3p ) - ah1**3 ) /    &
+           ( ah1**3 + K1p * ah1**2 + K1p * K2p * ah1 + K1p * K2p * K3p )
+      ab   = borat / ( 1. + ah1 / Kb )
+      aw   = Kw / ah1 - ah1 / ( 1. + sti / Ks1 )
+      ac   = ta + hso4 - sit * hsi - ab - aw + ft * hf - pt * hpo4
+      ah2o = SQRT((K1*dic_h2co3)**2 + 4.*ac*2.*K1*k2*dic_h2co3)
+      ah2  = (K1*dic_h2co3 + ah2o)/(2.*ac)
+      erel = ( ah2 - ah1 ) / ah2
+      if (abs( erel ).ge.eps) then
+        ah1 = ah2
+      else
+        exit iflag
+      endif
+    enddo iflag
+
+    dic_hco3  = Kh * K1 *      pco2 * 1e-6 / ah1
+    dic_co3   = Kh * K1 * K2 * pco2 * 1e-6 / ah1**2
+    tc_sat    = dic_h2co3 + dic_hco3 + dic_co3
+
+  end subroutine carchm_solve_dicsat
+
+end module mo_carchm
```

Comment:
Hi @TomasTorsvik and @JorgSchwinger , for now, I did not compare the content of the two added subroutines (I considered them as unchanged besides the indentation).

Reply:
They were totally unchanged except for the indentation and other formatting changes that applied to the other routines.